File: csharp/resistor-color/ResistorColor.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class ResistorColor
{
    private static List<string> _colors = new List<string> { "black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white" };

    public static int ColorCode(string color)
    {
        if (color is null)
        {
            throw new ArgumentNullException(nameof(color));
        }

        int number = _colors.FindIndex(e => e == color);
        if (number == -1)
        {
            throw new ArgumentException($"Unknown color: {color}");
        }

        return number;
    }

    public static string[] Colors()
    {
        return _colors.ToArray();
    }
}

File: javascript/resistor-color/resistor-color.js
const colorCodeMap = {
    "black": 0,
    "brown": 1,
    "red": 2,
    "orange": 3,
    "yellow": 4,
    "green": 5,
    "blue": 6,
    "violet": 7,
    "grey": 8,
    "white": 9
};

const COLORS = Object.keys(colorCodeMap);

function colorCode(color) {
    if (colorCodeMap[color] === undefined) {
        throw "Unknown color";
    }

    return colorCodeMap[color];
};

export { colorCode, COLORS };
